internal/yunikorn/event_handler.go

```diff
@@ -37,234 +34,100 @@ func (s *Service) handleEvent(ctx context.Context, ev *si.EventRecord) error {
-			// For now, we will use the Event.TimestampNano as the RequestTime but this is not factually correct
-			RequestTime: ev.GetTimestampNano(),
-		}
-		app.Requests = append(app.Requests, ask)
-	default:
-		// should be warning
-		logger.Warnf("unknown event EventChangeDetail type for an Event of type APP: %v",
-			ev.GetEventChangeDetail())
-	}
-}
-
-// Usually, we receive a SET event for an application when the application state changes.
-// Except for the REJECT state, which is received as a REMOVE event.
-func (s *Service) handleAppSetEvent(ctx context.Context, ev *si.EventRecord) {
-	logger := log.FromContext(ctx)
 
-	switch ev.GetEventChangeDetail() {
-	case si.EventRecord_APP_NEW:
-		app, err := s.client.GetApplication(ctx, "", "", ev.GetObjectID())
-		if err != nil {
-			logger.Errorf(
-				"could not get application info %s from scheduler: %v\nReceived Event: %v",
-				ev.GetObjectID(), err, ev)
-			return
-		}
-		if app == nil {
-			logger.Errorf(
-				"received an new application event but the application was not found in the scheduler: %s",
-				ev.GetObjectID())
+		if err := s.repo.InsertApplication(ctx, app); err != nil {
+			logger.Errorf("could not insert application: %v", err)
 			return
 		}
-		s.appMap[ev.GetObjectID()] = app
-	case si.EventRecord_APP_ACCEPTED,
-		si.EventRecord_APP_STARTING, si.EventRecord_APP_RUNNING,
-		si.EventRecord_APP_COMPLETING, si.EventRecord_APP_COMPLETED,
-		si.EventRecord_APP_FAILING, si.EventRecord_APP_FAILED,
-		si.EventRecord_APP_RESUMING, si.EventRecord_APP_EXPIRED:
-		state := si.EventRecord_ChangeDetail_name[int32(ev.GetEventChangeDetail())]
-		app, ok := s.appMap[ev.GetObjectID()]
-		if !ok || app == nil {
-			// should be warning
-			logger.Warnf("an application state change of type %s was "+
-				"received for an application without a previous ADD event: %s",
-				state, ev.GetObjectID())
-			return
-		}
-		app.StateLog = append(app.StateLog, &dao.StateDAOInfo{
-			Time:             ev.GetTimestampNano(),
-			ApplicationState: state,
-		})
-		app.State = state
-		// Insert the application into the DB once it is completed or failed
-		// should we include the EXPIRED state?
-		if ev.GetEventChangeDetail() == si.EventRecord_APP_COMPLETED ||
-			ev.GetEventChangeDetail() == si.EventRecord_APP_FAILED {
 
-			if err := s.repo.UpsertApplications(ctx, []*dao.ApplicationDAOInfo{app}); err != nil {
-				logger.Errorf("could not insert application into DB: %v", err)
-			}
-		}
-	default:
-		// should be warning
-		logger.Warnf("unknown event EventChangeDetail type for an Event of type APP: %v",
-			ev.GetEventChangeDetail())
+		return
 	}
-}
 
-func (s *Service) handleAppRemoveEvent(ctx context.Context, ev *si.EventRecord) {
-	logger := log.FromContext(ctx)
+	app, err := s.repo.GetLatestApplicationByApplicationID(ctx, daoApp.ApplicationID)
+	if err != nil {
+		logger.Errorf("could not get application by application id: %v", err)
+		return
+	}
```

Comment:
There could be the same `app_id` present in multiple partitions. In such cases, `s.repo.GetLatestApplicationByApplicationID` might return a different application than the one we intend to update. Did we decide on having a convention of maintaining a globally unique `app_id`?

Reply:
All applications belong to the default partition